state/protocol/protocol_state/common/base_statemachine.go

```diff
@@ -29,8 +29,8 @@ func NewBaseKeyValueStoreStateMachine(
 }
 
 // Build is a no-op by default. If a state machine needs to persist data, it should override this method.
-func (m *BaseKeyValueStoreStateMachine) Build() (*transaction.DeferredBlockPersist, error) {
-	return transaction.NewDeferredBlockPersist(), nil
+func (m *BaseKeyValueStoreStateMachine) Build() ([]storage.BlockIndexingBatchWrite, error) {
+	return make([]storage.BlockIndexingBatchWrite, 0), nil
```

Comment:
list object-oriented character. Additions to the return value will copy

Reply:
[addressed in Follow-up PR 7609](https://github.com/onflow/flow-go/pull/7609/files#diff-4114f49282c1800c5c7ce604c0cbac697e842e7f9371da1fe8e7f9bcb3d7577aR113)